app/_kong_plugins/openid-connect/index.md

```diff
@@ -37,6 +35,541 @@ search_aliases:
+`config.issuer` <br> *required* | This parameter tells the plugin where to find discovery information, and it is the only required parameter. You should set the value `realm` or `iss` on this parameter if you don't have a discovery endpoint. <br><br>This parameter doesn't have to match the URL of the `iss` claim in the access tokens being validated. To set URLs supported in the `iss` claim, use `config.issuers_allowed`.
+`config.auth_methods` | Decide what authentication grants to use with this plugin and configure the `config.auth_methods` field accordingly. To restrict the scope of potential attacks, the parameter should only contain the grants that you want to use. 
+`config.client_id` | In many cases, you need to specify `config.client_id`. Also, if your identity provider requires authentication, such as on a token endpoint, you will need to specify the client authentication credentials, for example `config.client_secret`.
+`config.audience_required` | If you are using a public identity provider, such as Google, you should limit the audience with `config.audience_required` to contain only your `config.client_id`. You may also need to adjust `config.audience_claim` in case your identity provider uses a non-standard claim (other than `aud`, as specified in JWT standard). This is important because some identity providers share public keys with different clients.
+`config.session_secret` | If you are running {{site.base_gateway}} in DB-less mode with declarative configuration and session cookie authentication, set `config.session_secret`. Leaving this parameter unset will result in every Nginx worker across your nodes encrypting and signing the cookies with their own secrets.
+
+For all available configuration parameters, see the 
+[OpenID Connect configuration reference](/plugins/openid-connect/reference/).
+
+## Authentication flows and grants
+
+The plugin supports several types of credentials and grants.
+When this plugin is configured with multiple grants or flows, there is a hardcoded search
+order for the credentials:
+
+1. [Session authentication](#session-auth-workflow)
+2. [JWT access token authentication](#jwt-access-token-auth-flow)
+3. [Kong OAuth token authentication](#kong-oauth-token-auth-flow)
+4. [Introspection authentication](#introspection-authentication-flow)
+5. [User info authentication](#user-info-auth-flow)
+6. [Refresh token grant](#refresh-token-grant-workflow)
+7. [Password grant](#password-grant-workflow) (username and password)
+8. [Client credentials grant](#client-credentials-grant-workflow)
+9. [Authorization code flow](#authorization-code-flow) (with client secret or PKCE)
+
+Once it finds a set of credentials, the plugin stops searching, and won't look for any further credential types.
+
+Multiple grants may share the same credentials. For example, both the password and client credentials grants can use 
+basic authentication through the `Authorization` header.
+
+### Authorization code flow
```

Comment:
later, i would make this whole section an include. My suggestion (for after demo) would be to simplify all of the flows into two sentences, then link off to a doc that has them in detail including the diagram. The sentences would explain at a high-level what I need to know, and the linked doc would give me this diagram.

Reply:
Will revisit after merge.